### PR TITLE
Silence deprecation warnings from concurrent-ruby

### LINKED ISSUE #3869: WARN -- Concurrent: [DEPRECATED] Java 7 is deprecated, please use Java 8.
I still get this. Can we maybe monkey patch to suppress this warning until we can go to Java 8 for LS?

```
[2015-09-03 17:13:38.733]  WARN -- Concurrent: [DEPRECATED] Java 7 is deprecated, please use Java 8.
Java 7 support is only best effort, it may not work. It will be removed in next release (1.0).
```

Config

```
input {
  stdin { }
}

filter {
  grok {
    match => {
      "message" => '%{IPORHOST:clientip} %{USER:ident} %{USER:auth} \[%{HTTPDATE:timestamp}\] "%{WORD:verb} %{DATA:request} HTTP/%{NUMBER:httpversion}" %{NUMBER:response:int} (?:-|%{NUMBER:bytes:int}) %{QS:referrer} %{QS:agent}'
    }
  }

  date {
    match => [ "timestamp", "dd/MMM/YYYY:HH:mm:ss Z" ]
    locale => en
  }

  geoip {
    source => "clientip"
  }

  useragent {
    source => "agent"
    target => "useragent"
  }
}

output {
  stdout { codec => rubydebug }
  elasticsearch_java {
    protocol => "node"
    hosts => "localhost"
  }
}
```

### PR DESCRIPTION
**Pull request by ph**
Concurrent-ruby is throwing warning when running under jdk7,
we have decided to silence theses errors unless you run logstash with
the `--debug` flag.

Fixes #3869

### CODE REVIEW

**Comment by suyograo** [comment]
@guyboertje review please?

File: lib/logstash/patches/silence_concurrent_ruby_warning.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# ref: https://github.com/ruby-concurrency/concurrent-ruby/blob/v0.9.1/lib/concurrent/configuration.rb#L284-L295
+#
+# This patch is only valid for 0.9.1
+if Concurrent::VERSION == "0.9.1" 
+  module Concurrent
+    module Concern
+      module Deprecation
+        include Concern::Logging
+
+        def deprecated(message, strip = 2)
```
Comment (jordansissel) [inline lib/logstash/patches/silence_concurrent_ruby_warning.rb]: ehhh, do we even need to log this?

The way I interpret the deprecated warning is the concurrent-ruby developers communicating with consumers of that library. This message means nothing to Logstash users, so I feel that we shouldn't even log it. Just totally silence it.

Thoughts? Any value in logging it vs ignoring it completely?

Comment (ph) [inline lib/logstash/patches/silence_concurrent_ruby_warning.rb]: Well, I have been **bite** in the past by `concurrent-ruby` updates, I would prefer a way to at least debug if something changes. what do you think?

**Comment by ph** [comment]
jenkins test this please

**Comment by guyboertje** [comment]
LGTM - @ph: simple, effective and good

File: lib/logstash/patches/silence_concurrent_ruby_warning.rb (hunk 1 above)
Comment (ph) [inline lib/logstash/patches/silence_concurrent_ruby_warning.rb]: @jordansissel did I answer your question?

**Comment by suyograo** [comment]
@ph I still see the warnings in 2.0. I merged this patch https://github.com/elastic/logstash/commit/b863ace5ab319950fa6db228d4f208d3dcb5e2bc. Can you take a look again please. Maybe we can completely remove the warning

**Comment by ph** [comment]
@suyograo oh boy, this patch isn't included in the 2.0 beta release. o_O

```
~/D/logstash-2.0.0-beta1 ❯❯❯ ls -l vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0.beta1-java/lib/logstash/patches/                                                               ⏎
total 48
-rw-r--r--@ 1 ph  staff  1174 16 Sep 10:41 bugfix_jruby_2558.rb
-rw-r--r--@ 1 ph  staff   948 16 Sep 10:41 bundler.rb
-rw-r--r--@ 1 ph  staff  1233 16 Sep 10:41 cabin.rb
-rw-r--r--@ 1 ph  staff  1315 16 Sep 10:41 profile_require_calls.rb
-rw-r--r--@ 1 ph  staff  1423 16 Sep 10:41 rubygems.rb
-rw-r--r--@ 1 ph  staff  3462 16 Sep 10:41 stronger_openssl_defaults.rb
~/D/logstash-2.0.0-beta1 ❯❯❯
``

**Comment by ph** [comment]
Strangely this file exist in our 2.0 branch?!

```
/t/logstash git:remotes/origin/2.0 ❯❯❯ ls -l lib/logstash/patches/
total 56
-rw-r--r--  1 ph  wheel  1174 17 Sep 09:46 bugfix_jruby_2558.rb
-rw-r--r--  1 ph  wheel   948 17 Sep 09:46 bundler.rb
-rw-r--r--  1 ph  wheel  1233 17 Sep 09:46 cabin.rb
-rw-r--r--  1 ph  wheel  1315 17 Sep 09:46 profile_require_calls.rb
-rw-r--r--  1 ph  wheel  1423 17 Sep 09:46 rubygems.rb
-rw-r--r--  1 ph  wheel  2038 17 Sep 09:46 silence_concurrent_ruby_warning.rb
-rw-r--r--  1 ph  wheel  3462 17 Sep 09:46 stronger_openssl_defaults.rb
/t/logstash git:remotes/origin/2.0 ❯❯❯
```

**Comment by ph** [comment]
@suyograo When you published the logstash-core gem where you up to date with the repository?

```
~/Downloads ❯❯❯ gem unpack logstash-core-2.0.0.beta1-java.gem
Unpacked gem: '/Users/ph/Downloads/logstash-core-2.0.0.beta1-java'
~/Downloads ❯❯❯ ls -l logstash-core-2.0.0.beta1-java/lib/logstash/patches/
total 48
-rw-r--r--  1 ph  staff  1174 17 Sep 09:50 bugfix_jruby_2558.rb
-rw-r--r--  1 ph  staff   948 17 Sep 09:50 bundler.rb
-rw-r--r--  1 ph  staff  1233 17 Sep 09:50 cabin.rb
-rw-r--r--  1 ph  staff  1315 17 Sep 09:50 profile_require_calls.rb
-rw-r--r--  1 ph  staff  1423 17 Sep 09:50 rubygems.rb
-rw-r--r--  1 ph  staff  3462 17 Sep 09:50 stronger_openssl_defaults.rb
~/Downloads ❯❯❯
```

**Comment by ph** [comment]
Applying the patch manually on logstash 2.0 beta 1 fixes the issue, so its only a matter of releasing a logstash-core that include the fix.

**Comment by suyograo** [comment]
Will get this as part of beta2. Closing

**Pull request closed without merging**